### DISCUSSION TITLE
[Android] Logcat error "PIXEL: ioctl err: 1"

### DISCUSSION THREAD

**Discussion by ravenfeld** [maplibre/maplibre-native#1998] [Q&A]
Hello, I'm asking for your help because I don't understand the problem.
Everything works fine, but my logcat gets killed by this log "PIXEL: ioctl err: 1" every 12 ms
Do you have any tips on how to get rid of this problem?
I've run the dialing code but I have no other problems.

Thanks in advance

**Comment by louwers** [comment]
Hi @ravenfeld. Could you provide more detailed reproduction steps?

**Comment by ravenfeld** [comment]
I just touch the card or do nothing if the locationComponent is activated. It's as if animating the map is causing problems internally in the logcat.

On one plus 6 I don't have the message but GC requests

**Reply by louwers** [reply]
What version are you using?

Can you provide a minimal version of your code that recreates the problem?

**Comment by ravenfeld** [comment]
https://github.com/ramani-maps/ramani-maps
Yes you can test the polygon example of this lib, or I thought it was to put in compose but even by redoing an application in xml same worry.

**Comment by chjaeggi** [comment]
any updates on this @louwers ?
I also have this issue when including a map in a composable `AndroidView` especially when scrolling around on the map for instance.
This causes my phone to get really hot ...

**Reply by ravenfeld** [reply]
For the moment no, I have the impression that it depends on the phones, some of which don't report the error.

**Reply by louwers** [reply]
I have not seen this issue myself (using Samsung S10 mostly). Please provide a way to reproduce the problem that does not depend on external libraries.

Also feel free to create a bug report instead of a discussion. Thanks!

**Reply by chjaeggi** [reply]
I will come up with a minimal sample.
In the meantime as heads-up, I didn't observe this issue in emulator (Pixel 6 API 33 and Pixel 7 Pro API34)

**Comment by chjaeggi** [comment]
Hey @louwers 
I finally found the time to come up with a sample

You will find a lot of the ioctl errors when you pinch to zoom or dragging around the map.


``` Kotlin
class MainActivity : ComponentActivity() {

    private fun setUpMapLibre(okHttpClient: OkHttpClient) {
        Mapbox.getInstance(this, null, WellKnownTileServer.MapLibre)
        HttpRequestUtil.setOkHttpClient(okHttpClient)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        val okHttpClient = OkHttpClient.Builder().build()
        setUpMapLibre(okHttpClient)

        super.onCreate(savedInstanceState)
        setContent {
            MaterialTheme {
                Surface(
                    modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background
                ) {
                    val context = LocalContext.current
                    val mapView = MapView(context).apply {
                        getMapAsync { map ->
                            map.apply {
                                setStyle("https://raw.githubusercontent.com/maplibre/demotiles/gh-pages/styles/osm-bright-gl-style/style.json") {
                                    LocationComponentActivationOptions.builder(context, it).build()
                                }
                            }
                        }
                    }
                    AndroidView(factory = { mapView })
                }
            }
        }
    }
}
```

**Reply by chjaeggi** [reply]
PS: This is the most basic sample I could provide.
For the dependencies you need to include:
add this to your `gradle.settings.kts`

```
    implementation("org.maplibre.gl:android-sdk:10.2.0")
    implementation("org.maplibre.gl:android-plugin-annotation-v9:2.0.1")
    implementation("com.squareup.okhttp3:okhttp:4.11.0")
```

**Reply by chjaeggi** [reply]
It even gets worse when updating the map with a location bubble or so by incorporation of location services and co.
as wished this is only the most basic sample app I could come up with
